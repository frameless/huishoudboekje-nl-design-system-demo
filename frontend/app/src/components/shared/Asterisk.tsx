import React from "react";
import {HStack, Text} from "@chakra-ui/react";
import {useTranslation} from "react-i18next";

const Asterisk = () => {
	const {t} = useTranslation();
	return (
		<HStack spacing={1}>
			<Text color={"red.500"}>*</Text>
			<Text>{t("forms.asterisk")}</Text>
		</HStack>
	);
};

export default Asterisk;