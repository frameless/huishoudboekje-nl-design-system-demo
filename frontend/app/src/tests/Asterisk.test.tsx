import {getByText, render} from "@testing-library/react";
import React from "react";
import Asterisk from "../components/shared/Asterisk";

jest.mock("react-i18next", () => require("./utils/mock-hooks").reactI18NextMock());

describe("Asterisk", () => {

	it("Renders asterisk component", () => {
		const {container} = render(<Asterisk />);

		expect(container?.innerHTML).toMatchSnapshot();

		const asterisk = getByText(container!, "*");
		expect(asterisk).toBeDefined();

		const text = getByText(container!, "forms.asterisk");
		expect(text).toBeDefined();

	});
});