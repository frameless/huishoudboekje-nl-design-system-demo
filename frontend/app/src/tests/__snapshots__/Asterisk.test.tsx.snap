// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asterisk Renders asterisk component 1`] = `"<div class=\\"chakra-stack css-193rmy8\\"><p class=\\"chakra-text css-zov4ca\\">*</p><p class=\\"chakra-text css-0\\">forms.asterisk</p></div>"`;
